NCS.DSS.EmploymentProgression/PostEmploymentProgression/Function/EmploymentProgressionPostTrigger.cs

```diff
@@ -0,0 +1,187 @@
+                return _httpResponseMessageHelper.BadRequest();
+            }
+
+            var ApimURL = _httpRequestHelper.GetDssApimUrl(req);
+            if (string.IsNullOrEmpty(ApimURL))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, "Unable to locate 'apimurl' in request header");
+                return _httpResponseMessageHelper.BadRequest();
+            }
+
+            if (!Guid.TryParse(customerId, out var customerGuid))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Unable to parse 'customerId' to a Guid: {customerId}");
+                return _httpResponseMessageHelper.BadRequest(customerGuid);
+            }
+
+            if (!await _resourceHelper.DoesCustomerExist(customerGuid))
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, "Bad request");
+                return _httpResponseMessageHelper.NoContent();
+            }
+
+            var isCustomerReadOnly = await _resourceHelper.IsCustomerReadOnly(customerGuid);
+
+            if (isCustomerReadOnly)
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Customer is readonly with customerId {customerGuid}, correlationId {correlationGuid}.");
+                return _httpResponseMessageHelper.Forbidden(customerGuid);
+            }
+
+            var doesEmploymentProgressionExist = _employmentProgressionPostTriggerService.DoesEmploymentProgressionExistForCustomer(customerGuid);
+            if (doesEmploymentProgressionExist)
+            {
+                _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Employment progression details already exists for customerId {customerGuid}, correlationId {correlationGuid}.");
+                return _httpResponseMessageHelper.Conflict();
+            }
+
+            _loggerHelper.LogInformationMessage(logger, correlationGuid, $"Attempt to get resource from body of the request correlationId {correlationGuid}.");
+
+            Models.EmploymentProgression employmentProgressionRequest;
+            try
+            {
+                employmentProgressionRequest = await _httpRequestHelper.GetResourceFromRequest<Models.EmploymentProgression>(req);
+            }
+            catch (Exception ex)
+            {
+                _loggerHelper.LogException(logger, correlationGuid, "Unable to retrieve body from req", ex);
+                return _httpResponseMessageHelper.UnprocessableEntity(JObject.FromObject(new { Error = ex.Message }).ToString());
+            }      
+
+            if (employmentProgressionRequest == null)
+            {
+                return _httpResponseMessageHelper.UnprocessableEntity();
+            }
+
+            _employmentProgressionPostTriggerService.SetIds(employmentProgressionRequest, customerGuid, touchpointId);
```

Comment:
move to the model for consistency. Model is the only place which updates the model

Reply:
Agreed to keep functionality in Service.